src/devtools/domains/history.js

```diff
@@ -33,12 +36,24 @@ class History {
       }
     }
 
-    return { ...newHistory, list }
+    return { ...newHistory, list, focused }
+  }
+
+  focus(_old, id) {
+    let focused = null
+    if (_old.focused === id) {
+      focused = _old.head
+    } else {
+      focused = id
+    }
```

Comment:
since you have the ternary check `send('detail', isFocused ? head : action.id)`, is this switch even necessary? or can this just be something like:
```js
focus(old, id) {
  return { ...old, focused: id }
}
```